server/src/main/java/com/google/sps/utils/StateCollectionBuilder.java

```diff
@@ -0,0 +1,116 @@
+package com.google.sps.utils;
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.SetOptions;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.sps.utils.Ad;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Description: StateCollectionBuilder reads in advertisements from the "ads" collection in Firestore.
+ *              Ad documents that are geo-targeted at the same state are grouped together in a state
+ *              subcollection. These subcollections are populated with advertiser documents. Each 
+ *              document contains a ads field, which is an ArrayList of primary keys for advertisement
+ *              documents in the "ads" collection.
+ * Author: Kira Toal
+ * Date: July 20, 2020
+ */ 
+public class StateCollectionBuilder {
+
+  private static final Set<String> VALID_GEO_TARGETS = new HashSet<>(Arrays.asList("Alabama", "Alaska", 
```

Comment:
I seem to recall you also putting this list in some other file.  Was that file in Java as well?  If so, make it a constant defined in its own file so you only need to update it in one place when Puerto Rico becomes the 51st state.